lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -622,6 +662,10 @@ class MongoClientInterface {
             return this.putObjectVerCase4(c, bucketName, objName, objVal,
                                           params, log, cb);
         }
+        if (params && params.query) {
+            return this.putObjectNoVerConditional(
```

Comment:
Is this only the `NoVer` variant because blobserver never supports versioning? What would happen if blobserver were to try updating/deleting a versioned object put by cloudserver?